neutrons/inspired
# Conda/Python Package Process Creation and CI

# Short description of the changes:
The code includes changes in order to create python wheel and conda package.
Also, the default file paths are updated.


# Long description of the changes:
The code adds:
- environment.yaml with conda dependencies
- pyproject.toml
- conda.recipe/meta.yaml 
- github actions to create and upload conda packages to anaconda
- folder reorganization and imports updates
- file paths for models/dataset to show default at /SNS/ locations
- VERSION removed

# Check list for the pull request

# Check list for the reviewer

# Manual test for the reviewer
Create and activate the environment
- conda env create
- conda activate inspired-dev

Python wheel
- python -m build --wheel --no-isolation
- check-wheel-contents dist/inspired-*.whl

Conda package
1. create a conda package

- cd conda.recipe
- echo "versioningit $(versioningit ../)"
- CHANNELS="--channel conda-forge --channel pytorch --channel pyg"
- VERSION=$(versioningit ../) conda mambabuild $CHANNELS --output-folder . .
- conda verify noarch/inspired-*.tar.bz2

2. install a local conda package

- conda install noarch/<inspired .tar.bz2 file>

3. start the application
- inspired

# References
[6827](https://ornlrse.clm.ibmcloud.com/ccm/resource/itemName/com.ibm.team.workitem.WorkItem/6827)
